remove docs for spark2 (#6183)
treeverse/lakeFS: docs/reference/spark-client.md

## Getting Started

> We are removing support for Spark 2 in the lakeFS metadata client.
> {: .note }
> 

Reviewer: ```suggestion
Please note that Spark 2 is no longer supported with the lakeFS metadata client.
{: .note }

```
